Microsoft.Toolkit.Uwp.UI.Controls/ImageEx/ImageExBase.cs

```diff
@@ -203,6 +205,22 @@ protected override Size ArrangeOverride(Size finalSize)
             return base.ArrangeOverride(finalSize);
         }
 
+        /// <summary>
+        /// from https://referencesource.microsoft.com/#WindowsBase/Base/System/Windows/Rect.cs,04390f584fcba8e4
+        /// </summary>
+        private static bool IntersectsWith(Rect rect1, Rect rect2)
```

Comment:
I know there's the [Intersect](https://docs.microsoft.com/en-us/dotnet/api/windows.foundation.rect.intersect) method, but that my be too much extra work, eh? We should probably just add this as an extension in the Microsoft.Toolkit.Uwp package. @Sergio0694 any other suggestions?

Reply:
Agreed, that one API in particular is also super weird as it does side effect on the calling instance for some reason. I think adding an `IntersectWith` extension to the `.Uwp` package makes perfect sense, would be a nice addition next to the new `Point` and `Rect` extensions we added already a while back 😄

Reply:
@h82258652 mind making this an extension method in a new `RectExtensions` class in our [extensions here](https://github.com/windows-toolkit/WindowsCommunityToolkit/tree/master/Microsoft.Toolkit.Uwp/Extensions)?

Reply:
@michael-hawker Added. Can you review the unit test? Especially the test cases. I'm worried about the test cases are enough.